src/pages/Overview/OverviewPage.tsx

```diff
@@ -620,6 +620,37 @@ export class OverviewPage extends React.Component<OverviewProps, State> {
           namespaceActions.push(enableAction);
         }
       }
+      if (serverConfig.kialiFeatureFlags.istioUpgradeAction) {
+        const upgradeAction = {
+          isGroup: false,
+          isSeparator: false,
+          isDisabled: !canWrite,
+          title: 'Upgrade to Newer Istio',
```

Comment:
Perhaps here I'd use the values from the "istio.io/rev" label.
It's a minor thing but I'd separate the labeling of the namespace with the "tags" [1] as there can be another level of indirect pointers here.

[1] https://istio.io/latest/docs/setup/upgrade/canary/